### PR TITLE
[IOTDB-1926]TsFile API supports query on aligned timeseries

### PR DESCRIPTION
**Pull request by choubenson**
TsFile Api supports query on aligned timeseries, which also support value filter on aligned timeseries when querying.

### CODE REVIEW

File: example/tsfile/src/main/java/org/apache/iotdb/tsfile/TsFileRead.java [hunk 1]
```diff
@@ -40,55 +41,55 @@
...
     }
-    System.out.println("------------");
+    System.out.println("----------------");
   }
 
   public static void main(String[] args) throws IOException {
 
-    // file path
-    String path = "test.tsfile";
+    // file path, the number of levels of the directory must be greater than 3
```
Comment (qiaojialin) [inline example/tsfile/src/main/java/org/apache/iotdb/tsfile/TsFileRead.java:58]: what's this for?

File: example/tsfile/src/main/java/org/apache/iotdb/tsfile/TsFileWriteAlignedWithTSRecord.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public static void main(String[] args) throws IOException {
       writeMeasurementScheams.add(measurementSchemas.get(0));
       writeMeasurementScheams.add(measurementSchemas.get(1));
       writeMeasurementScheams.add(measurementSchemas.get(2));
-      writeAligned(tsFileWriter, "root.sg.d1", writeMeasurementScheams, 1000000, 0, 0);
+      writeAligned(tsFileWriter, "root.sg.d1", writeMeasurementScheams, 100000, 0, 0);
```
Comment (qiaojialin) [inline example/tsfile/src/main/java/org/apache/iotdb/tsfile/TsFileWriteAlignedWithTSRecord.java:59]: why change this?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/AlignedTimeSeriesMetadata.java [hunk 3]
```diff
@@ -112,6 +112,9 @@ public void setSeq(boolean seq) {
       // only at least one sensor exits, we add the AlignedChunkMetadata to the list
       boolean exits = false;
       for (List<IChunkMetadata> chunkMetadata : valueChunkMetadataList) {
+        if (i >= chunkMetadata.size()) { // there may be fewer value chunks than time chunks
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/AlignedTimeSeriesMetadata.java:115]: no need

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java [hunk 4]
```diff
@@ -327,7 +329,7 @@ public BloomFilter readBloomFilter() throws IOException {
     return deviceMetadata;
   }
 
-  public TimeseriesMetadata readTimeseriesMetadata(Path path, boolean ignoreNotExists)
+  public ITimeSeriesMetadata readTimeseriesMetadata(Path path, boolean ignoreNotExists)
```
Comment (qiaojialin) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java:332]: this method is used by the server, where we know whether the timeseries are aligned or not. So no need to change this method you could write a new method (readITimeseriesMetadata) for TsFile API.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

**Pull request merged**